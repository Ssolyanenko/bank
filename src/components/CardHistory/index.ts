export * from './History';
export * from './Pagination';
export * from './CardHistory';
