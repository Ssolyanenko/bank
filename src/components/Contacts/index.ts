export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
