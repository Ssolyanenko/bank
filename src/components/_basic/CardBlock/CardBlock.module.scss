@import 'src/styles/variables.module.scss';

.cardBlockWrapper {
  display: flex;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  gap: 32px;

  .cardInfoWrapper {
    width: 439px;
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 24px;

    .cardInfoTitle {
      font-weight: $bold;
      font-size: 1.5rem;
      line-height: 2rem;
      text-transform: capitalize;
      color: $grayDark_6;
      margin: 0;
    }

    .cardInfo {
      width: 100%;
      padding-left: 25px;
      list-style-type: disc;
      font-weight: $normal;
      font-size: 1rem;
      line-height: 1.5;
      color: $grayDark_6;
    }

    .removeCircle {
      list-style-type: none;
      padding-left: 0;
    }

    .primaryButton {
      align-self: center;
    }

    .button {
      &:hover {
        cursor: pointer;
        background-color: $brand_6;
        border: 1px solid $brand_6;
      }

      &:disabled {
        cursor: not-allowed;
        background-color: $brand_4;
      }
    }
  }
}

.cardBlockStoryBook {
  width: 800px;
  justify-content: center;
  gap: 60px;
  margin: 0 auto;
}
