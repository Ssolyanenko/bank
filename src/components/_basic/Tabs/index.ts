export * from './Tabs';
export * from './TabTitle';
