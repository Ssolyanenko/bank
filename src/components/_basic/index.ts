export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
