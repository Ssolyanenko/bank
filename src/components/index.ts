export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
