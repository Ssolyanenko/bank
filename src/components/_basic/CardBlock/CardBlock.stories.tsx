import { ReactElement } from 'react';
import { ComponentMeta } from '@storybook/react';
import { MemoryRouter } from 'react-router';

import { CardTypes } from 'constants/cardTypes';
import { PaymentSystem } from 'constants/paymentSystem';
import { CardBlock, CardBlockProps } from 'components/_basic/CardBlock';
import { convertUnixTimestampToDate } from 'helpers';
import classes from './CardBlock.module.scss';

export default {
  title: 'CardTemplate/Block layout for Cards',
  component: CardBlock,
  argTypes: {
    cardBlockTitle: {
      type: 'string',
      defaultValue: 'Smart Debit card',
      options: ['Credit Card Billable', 'Credit Card Billable Premium', 'Smart Debit card', 'Debit Travel card'],
      control: { type: 'select' },
    },

    cardProductName: {
      type: 'string',
      defaultValue: 'Smart Debit card',
      options: ['Credit Card Billable', 'Credit Card Billable Premium', 'Smart Debit card', 'Debit Travel card'],
      control: { type: 'select' },
    },
    cardType: {
      type: 'string',
      defaultValue: CardTypes.DEBIT_SMART,
      description: 'Type of card',
      options: [CardTypes.BILLABLE, CardTypes.BILLABLE_PREMIUM, CardTypes.DEBIT_SMART, CardTypes.DEBIT_TRAVEL],
      control: { type: 'select' },
    },
    paymentSystem: {
      name: 'Visa/MasterCard',
      type: 'string',
      defaultValue: PaymentSystem.MASTER_CARD,
      description: 'Visa or MasterCard type',
      options: [PaymentSystem.MASTER_CARD, PaymentSystem.VISA],
      control: { type: 'radio' },
    },
    cardDateStoryBook: {
      type: 'number',
      description: 'The end date of card',
      defaultValue: 1663707600000,
      control: { type: 'date' },
    },
    hasButton: {
      type: 'boolean',
      description: 'Do we need button or not',
      defaultValue: true,
      options: [true, false],
      control: { type: 'radio' },
    },
    isCircle: {
      type: 'boolean',
      description: 'is Circle list or not',
      defaultValue: false,
      options: [true, false],
      control: { type: 'radio' },
    },
    cardNumber: {
      type: 'string',
      defaultValue: '7777',
      description: 'The last four digits of the card number',
    },
    productInfo: {
      type: 'string',
      description: 'card promotional information',
      defaultValue:
        'Free transfers by phone number up to 5,000 GBP|Up to 30% cashback in dollars for purchases|Free card service',
    },
    redirectPath: {
      type: 'string',
      description: 'temporary route path',
      defaultValue: 'somePath',
    },
    cvv: {
      type: 'string',
      defaultValue: '***',
      description: 'cvv code',
    },
  },
  parameters: {
    docs: {
      description: {
        component: 'Block layout for Cards',
      },
    },
  },
} as ComponentMeta<typeof CardBlock>;

interface Props extends CardBlockProps {
  cardDateStoryBook: number;
}

const Template = ({
  cardProductName,
  cardDateStoryBook,
  paymentSystem,
  cardType,
  cardNumber,
  productInfo,
  isCircle,
  hasButton,
  cvv,
}: Props): ReactElement => (
  <MemoryRouter>
    <CardBlock
      hasButton={hasButton}
      id={1}
      cardProductName={cardProductName}
      paymentSystem={paymentSystem}
      cardType={cardType}
      cvv={cvv}
      isCircle={isCircle}
      cardNumber={cardNumber}
      productInfo={productInfo}
      className={classes.cardBlockStoryBook}
      onClick={(): void => alert('Click button')}
      cardDate={convertUnixTimestampToDate(cardDateStoryBook)}
    />
  </MemoryRouter>
);

export const BlockForDebitSmart = Template.bind({});
