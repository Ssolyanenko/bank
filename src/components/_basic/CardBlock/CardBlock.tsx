import { FC, ReactElement } from 'react';
import i18next from 'i18n';

import { PrimaryButton, CardTemplate } from 'components/_basic';
import { toDo } from 'helpers';
import { CardTemplate as CardTemplateProps } from 'interfaces/cardTemplate';
import classes from './CardBlock.module.scss';

export interface CardBlockProps extends CardTemplateProps {
  hasButton: boolean;
  buttonText?: string;
  isCircle?: boolean;
  onClick?(): void;
  className?: string;
}

export const CardBlock: FC<CardBlockProps> = ({
  id,
  paymentSystem,
  cardProductName,
  amount,
  cardDate,
  productInfo,
  cardNumber,
  cvv,
  hasButton,
  buttonText = i18next.t('cardBlock.orderACard'),
  isCircle = false,
  className,
  onClick = toDo,
}): ReactElement => (
  <section key={id} className={`${classes.cardBlockWrapper} ${className}`}>
    <CardTemplate
      id={id}
      key={id}
      cardProductName={cardProductName}
      paymentSystem={paymentSystem}
      amount={amount}
      cardDate={cardDate}
      cardNumber={cardNumber}
      cvv={cvv}
    />
    <div className={classes.cardInfoWrapper}>
      {hasButton && <h5 className={classes.cardInfoTitle}>{cardProductName}</h5>}
      <ul className={`${classes.cardInfo} ${isCircle ? classes.removeCircle : ''} `}>
        {productInfo?.split('|').map(
          (infoItem: string, index: number): ReactElement => (
            <li key={index + infoItem}>{infoItem}</li>
          )
        )}
      </ul>
      {hasButton && (
        <PrimaryButton className={classes.primaryButton} onClick={onClick}>
          {buttonText}
        </PrimaryButton>
      )}
    </div>
  </section>
);
