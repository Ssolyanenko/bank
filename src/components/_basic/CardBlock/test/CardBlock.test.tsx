import 'jsdom-global/register';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

import { CardBlock } from 'components/_basic';
import { mockInitialState } from 'constants/mockInitialState';
import { mockStore } from 'helpers';
import { BlockForDebitSmart } from 'components/_basic/CardBlock/CardBlock.stories';
import { ButtonNames } from 'constants/text';
import { CardName } from 'interfaces/cardTemplate';

const props = {
  id: 1,
  paymentSystem: '',
  cardProductName: '' as CardName,
  amount: 0,
  cardDate: '',
  productInfo: '',
  cardNumber: '',
  cvv: '',
  hasButton: false,
  buttonText: ButtonNames.ORDER_CARD,
  isCircle: false,
  onClick: (): void => {},
};

describe('<CardBlock />', (): void => {
  test('should make a snapshot', (): void => {
    const store = mockStore(mockInitialState);
    const wrapper = shallow(
      <Provider store={store}>
        <BlockForDebitSmart cardDateStoryBook={0} {...props} />
      </Provider>
    );
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render CardBlock from storybook', (): void => {
    const wrapper = shallow(<BlockForDebitSmart cardDateStoryBook={0} {...props} />);

    expect(wrapper.find(CardBlock).length).toBe(1);
  });
});
