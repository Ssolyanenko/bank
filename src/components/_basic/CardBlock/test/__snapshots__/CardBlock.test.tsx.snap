// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBlock /> should make a snapshot 1`] = `
<Context.Provider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <bound Template
    amount={0}
    buttonText="Order a card"
    cardDate=""
    cardDateStoryBook={0}
    cardNumber=""
    cardProductName=""
    cvv=""
    hasButton={false}
    id={1}
    isCircle={false}
    onClick={[Function]}
    paymentSystem=""
    productInfo=""
  />
</Context.Provider>
`;
